packages/fxa-react/components/Portal/index.tsx

```diff
@@ -5,16 +5,48 @@ import './index.scss';
 
 type PortalProps = {
   id: string;
+  headerId?: string;
+  descId?: string;
   children: React.ReactNode;
 };
 
-const Portal = ({ id, children }: PortalProps): React.ReactPortal | null => {
+const setA11yOnAdjacentElementsAndBody = (els: NodeListOf<HTMLElement>) => {
+  document.body.classList.add('overflow-hidden');
+  els.forEach((el) => {
+    el.setAttribute('aria-hidden', 'true');
+    el.classList.add('pointer-events-none');
+  });
+};
+
+const resetA11yOnAdjacentElementsAndBody = (els: NodeListOf<HTMLElement>) => {
+  document.body.classList.remove('overflow-hidden');
+  els.forEach((el) => {
+    el.removeAttribute('aria-hidden');
+    el.classList.remove('pointer-events-none');
+  });
+};
+
+const Portal = ({
+  id,
+  headerId = '',
+  descId = '',
+  children,
+}: PortalProps): React.ReactPortal | null => {
   let el = document.getElementById(id);
   if (!el) {
     el = document.createElement('div');
     el.setAttribute('class', 'portal');
     el.setAttribute('id', id);
+    el.setAttribute('role', 'dialog');
     document.body.appendChild(el);
+
+    if (id === 'modal') {
+      el.setAttribute('aria-labelledby', headerId);
+      el.setAttribute('aria-describedby', descId);
+      setA11yOnAdjacentElementsAndBody(
+        document.querySelectorAll('body > div:not(#modal)')
```

Comment:
So, if I'm understanding this correctly, this will only really target `#root`, right?

Reply:
Right now, yes, and it'll likely stay that way, but I wanted to future-proof this in case something else is ever pulled outside of and set adjacent to `#root`.